server/routes/serviceProviderReferrals/dashboardPresenter.test.ts

```diff
@@ -22,93 +70,96 @@ describe(DashboardPresenter, () => {
+            { text: 'Rob Shah-Brookes', sortValue: 'shah-brookes, rob', href: null },
+            {
+              text: 'Accommodation Services - West Midlands',
+              sortValue: null,
+              href: null,
+            },
+            { text: 'UserABC', sortValue: null, href: null },
+            {
+              text: 'View',
+              sortValue: null,
+              href: '/service-provider/referrals/1/progress',
+            },
           ],
           [
-            { text: '13 Oct 2020', sortValue: '2020-10-13', href: null },
+            { text: '14 Oct 2020', sortValue: '2020-10-14', href: null },
             { text: 'ABCABCA2', sortValue: null, href: null },
-            { text: 'Jenny Jones', sortValue: 'jones, jenny', href: null },
+            { text: 'Hardip Fraiser', sortValue: 'fraiser, hardip', href: null },
             { text: "Women's Services - West Midlands", sortValue: null, href: null },
             { text: '', sortValue: null, href: null },
-            { text: 'View', sortValue: null, href: `/service-provider/referrals/2/details` },
+            {
+              text: 'View',
+              sortValue: null,
+              href: '/service-provider/referrals/2/details',
+            },
+          ],
+          [
+            { text: '13 Oct 2020', sortValue: '2020-10-13', href: null },
+            { text: 'ABCABCA3', sortValue: null, href: null },
+            { text: 'Jenny Catherine', sortValue: 'catherine, jenny', href: null },
+            {
+              text: 'Accommodation Services - West Midlands',
+              sortValue: null,
+              href: null,
+            },
+            { text: 'UserABC', sortValue: null, href: null },
+            {
+              text: 'View',
+              sortValue: null,
+              href: '/service-provider/referrals/3/progress',
+            },
           ],
         ])
       })
 
       describe('when a referral has been assigned to a caseworker', () => {
         it('includes the caseworker’s username', () => {
-          const referralsSummary: ServiceProviderSentReferralSummary[] = [
-            {
-              referralId: '1',
-              sentAt: '2021-01-26T13:00:00.000000Z',
-              referenceNumber: 'ABCABCA1',
-              interventionTitle: 'Accommodation Services - West Midlands',
-              assignedToUserName: 'john.smith',
-              serviceUserFirstName: 'George',
-              serviceUserLastName: 'Michael',
-            },
-          ]
-          const presenter = new DashboardPresenter(referralsSummary, dashboardType, loggedInUser)
+          const page = pageFactory.pageContent(referrals).build() as Page<SentReferral>
+          const presenter = new DashboardPresenter(page, dashboardType, loggedInUser, interventions)
 
-          expect(presenter.tableRows[0][4]).toMatchObject({ text: 'john.smith' })
+          expect(presenter.tableRows[0][4]).toMatchObject({ text: 'UserABC' })
```

Comment:
This condition feels a bit magic-y; the previous version explicitly declared `assignedToUserName` but the new data setup does not -- this might condition might pass on all rows

Reply:
i think a counter condition - that an unassigned referral does not show the name on a dashboard like 'all cases' - would be sufficient to validate the test case.